plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduSecurityModule.java

```diff
@@ -106,6 +112,7 @@ private static KuduClientSession createKuduClientSession(KuduClientConfig config
             builder.disableStatistics();
         }
         KuduClientWrapper client = kuduClientFactory.apply(builder);
+        resourceManager.addInstance(client);
```

Comment:
Isn't annotation PreDestroy enough to close it when runner is closed?

Reply:
Aha - I see you're right: https://github.com/airlift/airlift/blob/master/docs/ref_lifecycle.md
(from `trino-raptor-legacy` - I saw "@PreDestroy" annotation was required because `lifeCycleManager.addInstance(kuduClientWrapper)` would ignore the instance if the annotation wasn't present. I didn't see the connection between Bootstrap + javax annotations 👍 )
```
        BackupStore proxy = new TimeoutBackupStore(
                store,
                catalogName.toString(),
                config.getTimeout(),
                config.getTimeoutThreads());

        lifeCycleManager.addInstance(proxy);
```
(^ probably could be removed, based on what Bootstrap guarantees?)

Reply:
Removed changes related to LifeCycleManager + kept @PreDestrory + explicit javax.annotations-api dependency.

Reply:
Nice, I didn't know this was taken care of by airlift.

Reply:
I think it relies on the `Bootstrap.initialize()` iterating over all `managedInstances` - and without a `@Provider KuduClientWrapper` in one of the modules, the singleton KuduClientWrapper won't be destroyed.
```
            this.log.info("Life cycle starting...");
            Iterator var1 = this.managedInstances.iterator();
```
We could add it to `KuduClientSession` (and make it `KuduClientSession`'s responsibility to close the underlying `KuduClientWrapper` - I'll see if that also has the desired effect.

Reply:
👍 adding a `@PreDestroy` to `KuduClientSession` and invoking `this.client.close();` has the desired effect w/out invoking the LifeCycleManager explicitly